pkg/kubectl/cmd/update.go

```diff
@@ -116,12 +116,13 @@ func updateWithPatch(cmd *cobra.Command, args []string, f *Factory, patch string
 	obj, err := helper.Get(namespace, name)
 	checkErr(err)
 
-	cmdutil.Merge(obj, patch, mapping.Kind)
+	patchedObj, err := cmdutil.Merge(obj, patch, mapping.Kind)
+	checkErr(err)
 
-	data, err := helper.Codec.Encode(obj)
+	data, err := helper.Codec.Encode(patchedObj)
 	checkErr(err)
 
-	obj, err = helper.Update(namespace, name, true, data)
+	_, err = helper.Update(namespace, name, true, data)
```

Comment:
Thanks for clarifying, I just wanted to double check that this wasn't supposed to be fixing everything we talked about in the comments of #4008, just the original issue.

I got interrupted last night when going through this, I'll check out the rest now.